docs/documentation/sql_metrics.md

```diff
@@ -1,11 +1,136 @@
+| ``avg_length`` | The average length of a string.  |  -  |
+| ``distinct`` |  The distinct contents in rows in a column.  | -  |
+| ``unique_count`` | The number of rows in which the content appears only once in the column.  |  - |
+| ``duplicate_count`` | The number of rows that contain duplicated content. | -  |
+| ``uniqueness`` | A measure of whether the rows contain unique content.  | -  |
+| ``maxs`` |  The number of rows that qualify as maximum. | ``valid_max`` |
+| ``mins`` |  The number of rows that qualify as minimum. | ``valid_min``  |
+| ``frequent_values`` |  The number of rows that contain content that most frequently occurs in the column. |  - |
+| ``histogram`` |  A histogram calculated on the content of the column.  | -  |
+
+| Column configuration key  | Description      | 
+| ---------- | ---------------- | 
+| ``metrics`` | Specifies extra metrics that Soda SQL computes for this column. |
+| ``metric_groups`` | Specifies extra metric groups that Soda SQL computes for this column. |
+| ``tests`` | A section that contains the tests that Soda SQL runs on a column during a scan.|
+| ``missing_values`` | Specifies the values that Soda SQL is to consider missing.|
+| ``missing_format`` | Specifies missing values such as whitespace or empty strings.|
+| ``missing_regex`` | Use regex expressions to specify your own custom missing values.|
+| ``valid_format`` | Specifies valid values with a named valid text format.|
+| ``valid_regex`` | Use regex exppressions to specify your own custom valid values. |
+| ``valid_values`` | Specifies several valid values in list format. |
+| ``valid_min`` | Specifies a min value for valid values. |
+| ``valid_max`` | Specifies a max value for valid values. |
+| ``valid_min_length`` | Specifies a min length for valid values. |
+| ``valid_max_length`` | Specifies a max length for valid values. |
+
+
+### Example tests using a column metric
+
+* ``invalid_percentage == 0`` in column ``id`` with column configuration ``valid_format: uuid`` checks the rows in the column named ``id`` for values that match a uuid (universally unique identifier) format. If the test passes, it means that 0% of the rows contain data that is invalid; if the test fails, it means that more than 0% of the rows contain invalid data, which is data that is in non-UUID format. 
+
+
+## Metric groups and dependencies
+
+By default, Soda SQL groups some metrics together because they depend on the same query. When you define a test that uses a metric that is part of a **metric group**, Soda SQL computes all the other metrics in that group during a scan. 
+
+| Metric groups |
+| ------------- |
+| ``missing_count`` <br /> ``missing_percentage`` <br /> ``values_count`` <br /> ``values_percentage`` |
+| ``valid_count`` <br /> ``valid_percentage`` <br /> ``invalid_count`` <br /> ``invalid_percentage`` |
+| ``distinct`` <br /> ``unique_count`` <br /> ``uniqueness`` <br /> ``duplicate_count`` |
```

Comment:
This part that explains how single metrics can trigger a group of metrics should be removed. We want to change this automagical resolving of metrics in a group to an explicit property.

There is already a separate `metric_groups` property for that now.  Can you figure it out from the code and tests?  See TestSodaServerInteraction.test_soda_server_client

Reply:
Ah, OK, got it. Yes, I'll take a look. If I have any difficulty, I'll ask!

Reply:
I do believe I figured it out. I am boldly moving ahead with merge.